cumulus/client/consensus/aura/src/collators/mod.rs

```diff
@@ -169,25 +170,33 @@ async fn can_build_upon<Block: BlockT, Client, P>(
 ) -> Option<SlotClaim<P::Public>>
 where
 	Client: ProvideRuntimeApi<Block>,
-	Client::Api: AuraApi<Block, P::Public> + AuraUnincludedSegmentApi<Block>,
+	Client::Api: AuraApi<Block, P::Public> + AuraUnincludedSegmentApi<Block> + ApiExt<Block>,
 	P: Pair,
 	P::Public: Codec,
 	P::Signature: Codec,
 {
 	let runtime_api = client.runtime_api();
 	let authorities = runtime_api.authorities(parent_hash).ok()?;
-	let author_pub = aura_internal::claim_slot::<P>(slot, &authorities, keystore).await?;
+	let author_pub = aura_internal::claim_slot::<P>(para_slot, &authorities, keystore).await?;
+
+	let Ok(Some(api_version)) =
+		runtime_api.api_version::<dyn AuraUnincludedSegmentApi<Block>>(parent_hash)
+	else {
+		return Some(SlotClaim::unchecked::<P>(author_pub, para_slot, timestamp))
```

Comment:
shouldn't we return `None` if we couldn't get the api version?

Reply:
Yeah, if we have `None`, it means the api doesn't exist and we should return an error.

Reply:
I changed this to be backward compatible with the previous behaviour. If we are building on top of the included block, the previous implementation would return `Some`. I don't think it is likely that someone runs this with a runtime without `AuraUnincludedSegmentApi`, but technically its possible.